src/components/segmentedControl/segmentedControl.svelte

```diff
@@ -0,0 +1,225 @@
+    // to see change state and allowing the user to manually close when ready. In other words, when we want
+    // the styles and menu navigation features of leo-menu-item, but we don't want the auto-close.
+
+    if (item.tagName === 'LEO-SEGMENTED-ITEM') {
+      selected = getValue(item)
+
+      dispatch('select-item', {
+        value: selected
+      })
+    }
+
+    // When using keyboard navigation to call selectItem, ensure click handlers happen on the item,
+    // but not when this was called in a click handler, otherwise we'll get 2x clicks.
+    if (item.tagName === 'LEO-SEGMENTED-ITEM' && e.type !== 'click') {
+      item.click()
+    }
+  }
+
+  let pill: HTMLDivElement;
+
+  onMount(() => {
+    pill.addEventListener("transitionstart", () => {
+      segmentedControl.classList.add("transitioning");
+    });
+
+    pill.addEventListener("transitionend", () => {
+      segmentedControl.classList.remove("transitioning");
+    });
+  });
+</script>
+
+<div
+  bind:this={segmentedControl}
+  class="leo-segmented-control"
+  class:isSmall={size === 'small'}
+  class:isTiny={size === 'tiny'}
+  on:keypress={(e) => {
+    if (e.code !== 'Enter' && e.code !== 'Space') return
+    selectItem(e)
+  }}
+  on:click={selectItem}
+>
+  <div class="pill" style:width={`${pillWidth}px`} style:left={`${pillPosition}px`} bind:this={pill} />
+  <slot />
+</div>
+
+<style lang="scss">
+  .leo-segmented-control {
+    --leo-icon-size: var(--leo-icon-m);
+    --bg: var(--leo-color-container-highlight);
+    --control-padding: var(--leo-spacing-s);
+    --item-padding: var(--leo-spacing-xl);
+    --gap: var(--leo-spacing-s);
+    --control-height: 48px;
+
+    display: flex;
+    max-width: max-content;
+    align-items: center;
+    position: relative;
+
+    background: var(--leo-segmented-control-bg, var(--bg));
+    padding: var(--control-padding);
+    gap: var(--leo-segmented-control-gap, var(--gap));
+    height: var(--leo-segmented-control-height, var(--control-height));
+    border-radius: var(--leo-segmented-control-height, var(--control-height));
+
+    &.isSmall {
+      --leo-icon-size: var(--leo-icon-s);
+      --control-height: 40px;
+      --item-padding: var(--leo-spacing-l);
+    }
+
+    &.isTiny {
+      --leo-icon-size: var(--leo-icon-xs);
+      --control-padding: var(--leo-spacing-xs);
+      --gap: var(--leo-spacing-xs);
+      --control-height: 28px;
+      --item-padding: var(--leo-spacing-m);
+    }
+
+    .pill {
+      background: var(--leo-color-container-background);
+      position: absolute;
+      height: calc(100% - (var(--control-padding) * 2));
+      min-width: var(--control-height);
+      border-radius: calc(var(--control-height) - var(--control-padding));
+      transition: width 0.2s cubic-bezier(0.22, 1, 0.36, 1), left 0.4s cubic-bezier(0.22, 1, 0.36, 1);
+    }
+
+    :global :where(&) ::slotted(leo-segmented-item),
+    :global :where(&) > leo-segmented-item {
```

Comment:
Just confirming that you've checked everything is all good in WebComponent land?